### ISSUE TITLE
viewWillAppear and viewDidAppear should be called on the menu view controller

### ISSUE THREAD

**Issue by alejandro-isaza** [romaonthego/RESideMenu#174]
When I call `presentLeftMenuViewController` I expect `viewWillAppear:` and `viewDidAppear:` to be called on the `leftMenuViewController`. Similarly with `rightMenuViewContoller` and on disappear. Currently the view appear methods are only called once when the app launches.

**Comment by airdrummingfool** [comment]
I agree that `viewDidAppear:` and `viewWillAppear:` should be called when revealing the menus. In my testing, they are called once, and the one time they are called is not even a time where the menu is coming into view.

**Comment by x2on** [comment]
Actually #227 doesn't cover all situations. If the menu is in front and you close it with gesture `viewDidAppear` from the `contentViewController` ist not called.

**Comment by airdrummingfool** [comment]
@x2on: If you're referring to events not being called on the content view controller when the menu opens or closes, that was a conscious decision. My current understanding is that the content view controller never fully leaves the screen when the menus are activated, so in my opinion it doesn't actually "disappear" or "appear". That interpretation is certainly up for debate/correction :)

As an alternative, you can use the `RESideMenuDelegate` methods such as `sideMenu:didHideMenuViewController:` to track when the content view controller is shifted away or restored to the center.

**Comment by x2on** [comment]
Ok now i get your purpose.

For me if the menu is shown the `contentViewController` is then disappeared, because you are focused on the menu and not you see only a small part of the `contentViewController` and can't interact with it.

I created the #236 to track if the `contentViewController` is changed for implementing the transition in the delegate.
